mutiny-core/src/vss.rs

```diff
@@ -0,0 +1,273 @@
+#![allow(dead_code)]
+use crate::auth::MutinyAuthClient;
+use crate::{error::MutinyError, logging::MutinyLogger};
+use aes::cipher::block_padding::Pkcs7;
+use aes::cipher::{BlockDecryptMut, BlockEncryptMut, KeyIvInit};
+use aes::Aes256;
+use anyhow::anyhow;
+use bitcoin::secp256k1;
+use bitcoin::secp256k1::SecretKey;
+use cbc::{Decryptor, Encryptor};
+use lightning::log_error;
+use lightning::util::logger::*;
+use reqwest::{Method, Url};
+use serde::{Deserialize, Serialize};
+use serde_json::{json, Value};
+use std::sync::Arc;
+
+type Aes256CbcEnc = Encryptor<Aes256>;
+type Aes256CbcDec = Decryptor<Aes256>;
+
+pub struct MutinyVssClient {
+    auth_client: Arc<MutinyAuthClient>,
+    url: String,
+    encryption_key: SecretKey,
+    logger: Arc<MutinyLogger>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct VssKeyValueItem {
+    pub key: String,
+    pub value: Option<Value>,
+    pub version: u32,
+}
+
+impl VssKeyValueItem {
+    /// Encrypts the value of the item using the encryption key
+    /// and returns an encrypted version of the item
+    pub(crate) fn encrypt(self, encryption_key: &SecretKey) -> EncryptedVssKeyValueItem {
+        // should we handle this unwrap better?
+        let bytes = self.value.unwrap().to_string().into_bytes();
```

Comment:
Why an option for `value`? Should we just enforce this?

Reply:
Because we reuse it for scanning keys, can probably just make its own struct for that function